controllers/apply-next/form-router.js

```diff
@@ -33,92 +37,70 @@ function initFormRouter(formModel) {
-        res.locals.currentlyEditingId = res.locals.getSessionData(sessionKeys.editingId);
-        if (res.locals.currentlyEditingId) {
-            applicationData = await ApplicationService.getApplicationById(
-                sessionKeys.form,
-                res.locals.currentlyEditingId
-            );
-        }
-        res.locals.currentApplicationData = applicationData ? applicationData.application_data : false;
+        const application = currentEditingId
+            ? await applicationsService.getApplicationById(sessionKeys.form, currentEditingId)
+            : {};
+
+        const currentApplicationData = get('application_data')(application);
+        res.locals.currentApplicationTitle = get('application_title')(application);
+        res.locals.currentApplicationData = currentApplicationData;
 
         // Translate the form object for each request and populate it with current user input
-        res.locals.form = prepareForm(req.i18n.getLocale(), formModel, res.locals.currentApplicationData);
+        const form = enhanceForm(req.i18n.getLocale(), formModel, currentApplicationData);
 
-        // Share some useful form variables
-        res.locals.validation = res.locals.getSessionData(sessionKeys.validation);
+        res.locals.form = form;
+        res.locals.formTitle = form.title;
+        res.locals.formBaseUrl = req.baseUrl;
         res.locals.FORM_STATES = FORM_STATES;
-        res.locals.user = req.user;
+        res.locals.breadcrumbs = [{ label: form.title, url: req.baseUrl }];
 
-        res.locals.formBaseUrl = req.baseUrl;
-        res.locals.formTitle = 'Application form: ' + res.locals.form.title;
+        res.locals.user = req.user;
         res.locals.isBilingual = formModel.isBilingual;
         res.locals.enablePrompt = false; // Disable prompts on apply pages
         res.locals.bodyClass = 'has-static-header'; // No hero images on apply pages
-        res.locals.breadcrumbs = [{ label: res.locals.form.title, url: req.baseUrl }];
 
         next();
     });
 
     /**
      * Route: Start page
      */
+    async function renderStartPage(req, res, errors = []) {
+        const { startPage } = res.locals.form;
+        const applications = await applicationsService.getApplicationsForUser(req.user.userData.id, sessionKeys.form);
+
+        res.render(startPage.template, {
+            title: res.locals.formTitle,
+            applications: applications,
+            csrfToken: req.csrfToken(),
+            errors: errors
+        });
+    }
+
     router
         .route('/')
-        .all((req, res, next) => {
-            const { startPage } = res.locals.form;
-            if (!startPage) {
-                throw new Error('No startpage found');
-            }
-            if (!startPage.urlPath && !startPage.template) {
-                throw new Error('No valid startpage types found');
-            }
-            next();
-        })
-        .get(cached.noCache, async function(req, res) {
```

Comment:
I think we might still need the `cached.noCache` stuff here (possibly on every route?) as when I was using this on the TEST environment, I kept getting redirected to the login page even after logging in (eg. the 302 redirect from `/apply-next/simple` had been cached)?

Reply:
It is on all the routes:

```
router.use((req, res, next) => {
        req.session.redirectUrl = req.baseUrl;
        req.session.save(() => {
            next();
        });
    }, requireUserAuth);

    router.use(cached.csrfProtection, async (req, res, next) => {
        res.locals.setSessionData
```

`cached.csrfProtection` also applies `noCache`. **But** I think the redirect thing you are talking about it because of the order of the middleware above, which needs to come after the csrf/nocache rules. Will take a look at merging the middleware to fix that.